support/oidc-discovery-provider/main.go

```diff
@@ -39,17 +39,24 @@ func run(configPath string) error {
 	}
 	defer log.Close()
 
-	if config.Domain != "" {
-		log.Warn("The `domain` configurable is deprecated and will be removed in a future release; use `domains` instead.")
-	}
-
 	source, err := newSource(log, config)
 	if err != nil {
 		return err
 	}
 	defer source.Close()
 
-	var handler http.Handler = NewHandler(config.Domains, source, config.AllowInsecureScheme, config.Domain == "")
+	domainPolicy, err := DomainAllowlist(config.Domains...)
+	if err != nil {
+		return err
+	}
+	if config.Domain == "" {
```

Comment:
I'm not sure about this if.
If I understand it, we will fails in the start if `len(config.Domains) == 0 && config.Domain == ""`,
but use Domain || Domains if only on is provided, 
or error if both.
However this line suggest me that if `domain == ""`  but domains is set, we'll allow `All`.
am i missing something?

Reply:
I got this check backward 😬 It should be `if config.Domain != "" {`..... the scary thing is I tested it 😱